### DISCUSSION TITLE
Why does Visual Studio Code can only find some of the exports coming from @acme/ui?

### DISCUSSION THREAD

**Discussion by j-fdion** [t3-oss/create-t3-turbo#1157] [Q&A] [answered]
Some examples:

Alert: OK
![image](https://github.com/user-attachments/assets/ea774016-ead9-436a-9de7-6bcf0d7009e9)

Badge: OK
![image](https://github.com/user-attachments/assets/e7cd804d-120b-4d82-85a8-50daf9db3aa4)

AspectRatio: Not found
![image](https://github.com/user-attachments/assets/bda2a3b1-e2a2-475b-ab51-d1ad569c3cc5)

Card: Not found
![image](https://github.com/user-attachments/assets/315a63e0-c699-4802-9939-65811d1729e3)

I'm kinda confused on why it does since it's all in the same dir.

I can even import them fine:
![image](https://github.com/user-attachments/assets/18610224-ea7d-4dbe-aea3-576262411bb5)

**Comment by juliusmarminge** [comment, marked as answer]
This is due to how typescript handles wildcard exports condition. it wont try to scan it preemptively due to performance reasons, so any exports not yet in the program will be unknown to typescript (hence also vscode auto import)

when it's in the program you will get auato-import of it which is probably what's happening for you

**Reply by juliusmarminge** [reply]
if you want all exports to be discovered at all times you need to manually specify all of them in `package.json#exports` instead of using wildcard

https://github.com/t3-oss/create-t3-turbo/blob/8ca45cd2b06096c14d36a713dce32d7afcb1fed7/packages/ui/package.json#L11-L17

**Reply by j-fdion** [reply]
Thank you! Didn't expect I'd receive such a fast answer, It's really appreciated. Somehow I thought I had broken something inside my project.
